sdk/sync/redis_test.go

```diff
@@ -140,6 +140,32 @@ func TestBarrier(t *testing.T) {
 	}
 }
 
+func TestBarrierCancel(t *testing.T) {
+	ctx, cancel := context.WithTimeout(context.Background(), 5*time.Second)
+	defer cancel()
+
+	close := ensureRedis(t)
+	defer close()
+
+	runenv := randomRunEnv()
+
+	watcher := MustWatcher(ctx, runenv)
+	defer watcher.Close()
+
+	state := State("yoda")
+	ch := watcher.Barrier(ctx, state, 10)
+	cancel()
+	select {
+	case err := <-ch:
+		if err != context.Canceled {
+			t.Errorf("expected context canceled, got %s", err)
+		}
+	case <-time.After(3 * time.Second):
+		t.Error("expected a cancel")
+		return
+	}
+}
```

Comment:
This _doesn't_ test the change in this PR. I just added this test to make sure barrier cancellation actually works.

Reply:
Ok, so this _is_ actually broken. It looks like redis treats context like a suggestion, more than anything else.

Reply:
Nevermind, my test is broken.